### ISSUE TITLE
Let martsy submit analytics events for Eigen where applicable

### ISSUE THREAD

**Issue by alloy** [artsy/eigen#388]
We need to figure out what we need to do to track events across martsy views. If the analytics endpoint simply needs a UUID for the active user, then we can just pass that to all martsy calls and have martsy send the events to snowplow?

This would mean that martsy should also pass that parameter around when linking to other views, to ensure that this is maintained when Eigen doesn’t provide native views for those views being linked to.

@craigspaeth Does that sound like something that’s doable on your end?

----

```
                          [Eigen]
                             ↓
          /works-for-you?userTrackingID=f856b477
                             ↓
                       [Martsy/Force]
                             ↓
/events?name=works-for-you&client=Eigen&userTrackingID=f856b477
                             ↓
                        [Segment.io]
```

**Comment by ashfurrow** [comment]
@craigspaeth Any ideas on this? Would love to get underway. /cc @alexanderfabry

**Comment by orta** [comment]
I can see two ways to do this:
 * Martsy / force checks if it's hosted in eigen, then reports that in their own analytics
 * We expose an [analytics object](http://nshipster.com/wkwebkit/#message-handlers) to the webview, martsy/force checks for its existence then sends along it's events to our object where eigen sends all the analytics

**Comment by alloy** [comment]
@orta I’d really prefer the former, it would cut down on requests the app has to make (which is better from our mobile pov) and most infrastructure already exists (assuming martsy tracks these events as well).

The only thing we’d need to add is a way for martsy to track a user that is not signed in, e.g. generate a UUID and pass that along.

**Comment by ashfurrow** [comment]
Based on conversations with Ani, it seemed like they're looking for more of a funnel to track a user session from Native -> Martsy -> Native and so on. We would need to provide some sort of interaction between eigen/martsy then, wouldn't we?

**Comment by ashfurrow** [comment]
Ah, @alloy just said that. Ok.

**Comment by craigspaeth** [comment]
This sounds reasonable on the web end. Still kinda unclear exactly how we implement this. Right now the only thing we use to check against Eigen is the header. So we could use that to pass in a `in_native_app` param to our analytics calls. For logged out users maybe you can pass in another header like `x-eigen-session-id` and we can likewise pass that along analytics calls?

Anyways feel free to log an issue in Microgravity/Force explaining the details of what needs to be implemented on the web end, bring it up before user team priorities next Weds., and maybe this is something @jtotoole can work on next sprint.

**Comment by anipetrov** [comment]
Happy to close this now as microgravity segment events do carry context user agent properties which let us tag them as Eigen events.

**Comment by alloy** [comment]
@anipetrov Are you able to associate those events to specific Eigen users?

**Comment by anipetrov** [comment]
That's a good point @alloy -- when they are logged in it will be fine, but not when we have trial users. The thread is still valid then.

**Comment by alloy** [comment]
This is fixed by #1199.
